### PR TITLE
Fixes to prepare for CRAN submission

### PR DESCRIPTION
**Pull request by klmr**
* [x] Fix spurious note about “unused” R6 dependency
* [x] Fix obsolete links
* [x] Fix bundle size

Currently, the `inst/doc` folder is much too large, and causes CRAN to complain because the overall install size is > 5 MB.

### CODE REVIEW

**Comment by klmr** [comment]
One suggestion to reduce the output size of the documentation would be to run `pngcrush` on the images before including them into the HTML files, but that would make the vignette building more complicated. A potential solution is the ‘optout’ package: https://coolbutuseless.github.io/2020/04/03/introducing-optout-a-package-for-optimizing-graphics-output/

**Comment by banfai** [comment]
`utils-tidyeval.R` was generated by `usethis::use_tidy_eval()`, but that's obsolete now  
see https://usethis.r-lib.org/reference/use_tidy_labels.html
should we clean this up now 🧹  or make an issue to fix later 🚀 ?

**Comment by klmr** [comment]
@banfai 🧹 (you can’t react with arbitrary emojis to GitHub comments, unfortunately.)

**Comment by idavydov** [comment]
I think this looks great, thanks @klmr !

Regarding optout, I think it's not on CRAN. And it's not clear how it was used in ggpattern which is now on CRAN.

**Comment by banfai** [comment]
> @banfai 🧹 (you can’t react with arbitrary emojis to GitHub comments, unfortunately.)

oops 😄

File: R/utils-datatable.R [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+#' section "data.table in Imports but nothing imported".
 #'
-#' @examples
-#' \dontrun{
-#' dt <- data.table::data.table(a = 1, b = 2)
-#' dt[, a := NULL]
-#' }
+#' @keywords internal
+#' @usage NULL
+#' @format NULL
```
Comment (banfai) [inline R/utils-datatable.R:9]: @klmr @idavydov  how about this?

File: .Rbuildignore [hunk 2]
```diff
@@ -19,3 +21,5 @@
 
 # vignette cache
 ^vignettes/cached/.*\.Rmd$
+# large vignette excluded on CRAN
+^vignettes/plate_scoring_examples.Rmd$
```
Comment (idavydov) [inline .Rbuildignore:25]: @klmr do you know if this still will be built for the pkgdown?

**Review by idavydov — CHANGES_REQUESTED** [review]
this looks really great, thanks @klmr 

I just wanted to double check that all vignettes will be still build on github pages, otherwise amazing

File: .Rbuildignore (hunk 2 above)
Comment (klmr) [reply .Rbuildignore:25]: Yes: see [pkgdown.r-lib.org/reference/build_articles.html](https://pkgdown.r-lib.org/reference/build_articles.html#suppressing-vignettes:~:text=If%20you%20want%20articles%20that%20are%20not%20vignettes%2C%20either%20put%20them%20in%20subdirectories%20or%20list%20in%20.Rbuildignore.).

**Review by idavydov — APPROVED** [review]

**Comment by idavydov** [comment]
@banfai once you approve, I can merge

**Review by banfai — APPROVED** [review]

**Pull request merged**